plugins/train/model/_base.py

```diff
@@ -356,6 +358,27 @@ def reset_pingpong(self):
         self.compile_predictors(initialize=False)
         logger.debug("Reset models")
 
+    def set_networks_trainable(self):
+        """ Set model's networks trainable state prior to compiling """
+        logger.debug("Setting Networks trainable state")
+
+        train_encoder = self.config.get("encoder_trainable", True)
+        train_decoder_a = self.config.get("decoder_a_trainable", True)
+        train_decoder_b = self.config.get("decoder_b_trainable", True)
+
+        encoder = self.networks['encoder'].network
+        for layer in encoder.layers:
+            layer.trainable = train_encoder
```

Comment:
? This won't set any of the layers in encoder.layers as trainable. Iterating over a iterator won't change the underlying items. You have to index though the iterator to actually affect them.
```
for index, layer in enumerate(encoder.layers):
    encoder.layers[index] = train_encoder
```